### PR TITLE
KAFKA-5929: Save pre-assignment to file to avoid too long text to display when do topic partition reassign

### PR DESCRIPTION
**Pull request by uncleGen**
When do partition reassign
- before pr
Pre-assignment will be printed directly. It is not friendly when the text is too long.

- after pr
Pre-assignment will still be printed directly, but will be save to a file at the same time, naming with suffix ".rollback" of "reassignment-json-file". For example:

```
./kafka-reassign-partitions.sh --reassignment-json-file test.json ...
```
then we may get a file **test.json.rollback**

### CODE REVIEW

**Comment by uncleGen** [comment]
retest this please.

**Comment by uncleGen** [comment]
retest this please

**Comment by uncleGen** [comment]
retest this please

**Comment by uncleGen** [comment]
ready for review

**Comment by uncleGen** [comment]
What a pity

**Comment by uncleGen** [comment]
@lindong28 Could you please take a review?

**Comment by asfgit** [comment]
FAILURE
 8067 tests run, 5 skipped, 2 failed.
--none--

**Comment by asfgit** [comment]
FAILURE
 8067 tests run, 5 skipped, 2 failed.
--none--

**Comment by asfgit** [comment]
SUCCESS 
 8067 tests run, 5 skipped, 0 failed.
--none--

File: core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala [hunk 1]
```diff
@@ -193,10 +194,10 @@ object ReassignPartitionsCommand extends Logging {
     val interBrokerThrottle = opts.options.valueOf(opts.interBrokerThrottleOpt)
     val replicaAlterLogDirsThrottle = opts.options.valueOf(opts.replicaAlterLogDirsThrottleOpt)
     val timeoutMs = opts.options.valueOf(opts.timeoutOpt)
-    executeAssignment(zkUtils, adminClientOpt, reassignmentJsonString, Throttle(interBrokerThrottle, replicaAlterLogDirsThrottle), timeoutMs)
+    executeAssignment(zkUtils, adminClientOpt, reassignmentJsonString, reassignmentJsonFile, Throttle(interBrokerThrottle, replicaAlterLogDirsThrottle), timeoutMs)
   }
 
-  def executeAssignment(zkUtils: ZkUtils, adminClientOpt: Option[JAdminClient], reassignmentJsonString: String, throttle: Throttle, timeoutMs: Long = 10000L) {
+  def executeAssignment(zkUtils: ZkUtils, adminClientOpt: Option[JAdminClient], reassignmentJsonString: String, reassignmentJsonFile: String, throttle: Throttle, timeoutMs: Long = 10000L) {
```
Comment (lindong28) [inline core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala:200]: making reassignmentJsonFile an Option[String] is probably better than using empty to indicate that it is not used.

**Review by lindong28 — COMMENTED** [review]
The patch LGTM overall. Left one comment.

I think this is a improvement but not that necessary. Usually I would re-direct output via pipe to another file. Committer can decide whether to accept or not.

File: core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala (hunk 1 above)
Comment (uncleGen) [reply core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala:200]: get it, thanks for your review!

**Pull request closed without merging**